test/test_kernel.jl

```diff
@@ -224,6 +221,7 @@ end
         @test r  == [1f0, 4f0]
     end
 
+#= TBD
```

Comment:
Very similar infinite loop in recursive function call `replace_different_layout`.